### ISSUE TITLE
Auto-hiding the menu bar sometimes causes tiling to act wonky

### ISSUE THREAD

**Issue by brettneese** [ianyh/Amethyst#349]
After enabling auto-hiding the OSX menu bar on 10.11 (http://www.cio.com/article/2938496/os-x/auto-hide-the-menubar-in-os-x-el-capitan.html), sometimes Amethyst will only tile up to where the menu bar would be if it weren't hidden, so there'll just be a tiny slice of desktop wallpaper at the top of the screen.

Next time it happens I'll nab a screenshot and try to pay more attention to what causes this to happen but I suspect it has something to do with Amethyst tiling windows while the menu bar is unhidden and not re-tiling when it hides.

[Trello Card](https://trello.com/c/rIpkD0ZR/176-amethyst-auto-hiding-the-menu-bar-sometimes-causes-tiling-to-act-wonky)

**Comment by ianyh** [comment]
It's unfortunately more subtle than that. The last time I looked at it it was actually an issue with small frame variations. It's been a long time since I wrote the original code, but if I recall there were issues with the accessibility APIs for hanging window frames. Sometimes the frame isn't entirely respected which results in the frame being slightly off from what we expected, which could in turn cause us to try to keep resizing it causing some really weird stuttering. I had to add code that ignored very small frame changes. I think the tolerance was 20px, which I think might be the height of menu bar. It might be bigger than that, but I think the issue is that we go to resize it when the menubar disappears, but the frame change gets ignored because the variation is so small.
